mqlc/mqlc.go

```diff
@@ -1670,16 +1670,88 @@ func (c *compiler) postCompile() {
 			chunk, typ, ref := c.expandListResource(chunk, ref)
 			switch chunk.Id {
 			case "$one", "$all", "$none", "$any":
+				// default fields
 				ref = chunk.Function.Binding
-				chunk := code.Chunk(chunk.Function.Binding)
+				chunk := code.Chunk(ref)
 				typ = types.Type(chunk.Function.Type)
-				c.expandResourceFields(chunk, typ, ref)
-				block.Datapoints = append(block.Datapoints, block.TailRef(ref))
+				err := c.expandResourceFields(chunk, typ, ref)
+				// when no defaults are definied or query isn't about a resource, no block was added
+				if err == nil {
+					block.Datapoints = append(block.Datapoints, block.TailRef(ref))
+
+					c.addValueFieldChunks()
+				}
 			default:
-				c.expandResourceFields(chunk, typ, ref)
+				// ignore errrors here, log messages are already printed
```

Comment:
i'm not seeing where they are printed

Reply:
They were printed inside `expandResourceFields`. But with the bool as return value, this changed anyways.